boards/FSM/Src/Io/Io_FlowMeters.c

```diff
@@ -0,0 +1,82 @@
+#include "main.h"
+#include "Io_SharedFreqOnlyPwmInput.h"
+#include "Io_FlowMeters.h"
+#include <assert.h>
+
+struct FlowMeters_Context
+{
+    struct FreqOnlyPwmInput *pwm_input;
+    TIM_HandleTypeDef *      htim;
+    HAL_TIM_ActiveChannel    active_channel;
```

Comment:
Isn't this really just the same as `rising_edge_tim_channel`? I don't understand why we have duplicated information here.

Reply:
`rising_edge_tim_channel` is used to indicate the channel when we are initializing the pwm input capture, but active channel is used to indicate the channel that we are detecting the interrupt from.

Reply:
Ok I see, but I still don't think this belongs in a separate struct. Any frequency only pwm input would need this information, not just flow meter, so it should just be in `struct FreqOnlyPwmInput`.

Reply:
You probably want to put htim and active_channel into `FreqOnlyPwmInput` and write getter functions for them. 

In the end, it should be 
```
static struct FreqOnlyPwmInput* primary_flow_meter, secondary_flow_meter;
```
and when you need `htim`

```
void Io_FlowMeters_InputCaptureCallback(TIM_HandleTypeDef *htim)
{
    if (htim == freqOnlyPwmInput_GetTimer(primary_flow_meter) &&
        htim->Channel == freqOnlyPwmInput_GetTimerChannel(primary_flow_meter))
    {
        Io_FlowMeters_TickPrimary();
    }
}
```

Reply:
Yeah using the getter makes it a lot cleaner. Fixed it. This should also address the comments above.